gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -358,8 +360,47 @@ MacroBuiltin::include_str_handler (Location invoc_locus,
 
   std::vector<uint8_t> bytes = load_file_bytes (target_filename.c_str ());
 
-  /* FIXME: Enforce that the file contents are valid UTF-8.  */
-  std::string str ((const char *) &bytes[0], bytes.size ());
+  /* unoptimized check for valid UTF-8 */
+  int expect_single = 0;
+  for (uint8_t b : bytes)
+    {
+      if (expect_single)
+	{
+	  if ((b & 0xC0) != 0x80)
+	    {
+	      /* character was truncated, exit with expect_single != 0 */
+	      break;
+	    }
```

Comment:
GCC styling convention wants us to remove these brackets
```suggestion
	      /* character was truncated, exit with expect_single != 0 */
	      break;
```